thorlcr/activities/lookupjoin/thlookupjoinslave.cpp

```diff
@@ -2496,11 +2503,22 @@ class CLookupJoinActivityBase : public CInMemJoinBase<HTHELPER, IHThorHashJoinAr
                  * If any have spilt, still need to distribute rest of RHS..
                  */
                 bool ok = handleGlobalRHS(marker, helper->isRightAlreadySorted(), stopping);
-                if (stopping)
+
+                if (queryJob().queryJobChannels()>1)
                 {
-                    ActPrintLog("Global getRHS stopped");
-                    return;
+                    if (stopping)
+                        channels[0]->queryRhsChannelStopSet()->set(queryJobChannelNumber(), true);
+                    InterChannelBarrier();
+                    // if all stopping, then scan will == # channels
+                    if (queryJob().queryJobChannels() == channels[0]->queryRhsChannelStopSet()->scan(0, false))
+                    {
+                        ActPrintLog("All channels in global getRHS stopped");
+                        return;
+                    }
                 }
+                else if (stopping)
+                    ActPrintLog("Global getRHS stopped");
```

Comment:
no longer returning.  Is that intentional?

Reply:
no, well spotted, will fix.